website/project/model.py

```diff
@@ -3621,81 +3681,97 @@ class DraftRegistration(AddonModelMixin, StoredObject):
     # have already been archived.
     # storage = fields.ForeignField('osfstoragenodesettings')
 
-    # Dictionary field mapping
-    # { 'requiresApproval': true, 'fulfills': [{ 'name': 'Prereg Prize', 'info': <infourl> }]}
-    # config = fields.DictionaryField()
+    approval = fields.ForeignField('draftregistrationapproval', default=None)
+
+    # Dictionary field mapping extra fields defined in the MetaSchema.schema to their
+    # values. Defaults should be provided in the schema (e.g. 'paymentSent': false),
+    # and these values are added to the DraftRegistration
+    flags = fields.DictionaryField()
 
-    # Dictionary field mapping a draft's states during the review process to their value
-    # { 'isApproved': false, 'isPendingReview': false, 'paymentSent': false }
-    # flags = fields.DictionaryField()
+    notes = fields.StringField()
 
     def __init__(self, *args, **kwargs):
         super(DraftRegistration, self).__init__(*args, **kwargs)
-        # TODO (samchrisinger): uncomment when we have flags/config
-        #meta_schema = self.registration_schema  # or kwargs.get('registration_schema')
-        #if meta_schema:
-        #    schema = meta_schema.schema
-        #    config = schema.get('config', {})
-        #    self.config = config
-        # TODO: uncomment to set flags
-        #     if not self.registration_schema:
-        #         flags = schema.get('flags', {})
-        #         for flag, value in flags.iteritems():
-        #             self.flags[flag] = value
-
-    # TODO: uncomment to expose approval/review properties
-    #    @property
-    #    def is_pending_review(self):
-    #        return self.flags.get('isPendingReview')
-    #
-    #    @is_pending_review.setter
-    #    def is_pending_review(self, value):
-    #        self.flags['isPendingReview'] = value
-    #
-    #    @property
-    #    def is_approved(self):
-    #        self.flags.get('isApproved')
-    #
-    #    @is_approved.setter
-    #    def is_approved(self, value):
-    #        self.flags['isApproved'] = value
+        meta_schema = self.registration_schema or kwargs.get('registration_schema')
+        if meta_schema:
+            schema = meta_schema.schema
+            if not self.registration_schema:
+                flags = schema.get('flags', {})
+                for flag, value in flags.iteritems():
+                    self.flags[flag] = value
```

Comment:
Looking at other usages of DictField, it seems that you may be able to use `self.flags.update(flags)` instead.